### PR TITLE
Slack-Apps Bitcamp Review #1

### PR DESCRIPTION
**Pull request by beaubhp**
Bahburs/BitCamp --> bitprj/BitCamp (Slack Apps Bitcamp Review)

### CODE REVIEW

File: Slack-Apps/week1/homework/responses/1.1-bolt-intro.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Introduction to Bolt JS
+
+Bolt JS is a framework, or a platform, for developing in JavaScript for Slack. Bolt **significantly** reduces the learning curve for Slack development by abstracting away some of the more complicated language barriers involved. In this issue, you will be familiarizing yourself with Bolt.
+
+## 📚 Reading Up
+
+Before getting into Bolt, [read up on Slack](https://api.slack.com/start/overview) and its use cases. This will help you get a better idea of what the projects that we will use Bolt to create. After you are done reading, you can start getting more into the technical details. Read [this article](https://api.slack.com/tools/bolt) on Bolt, then go through 2 or 3 code samples of your choice. The code samples can be found at the very bottom of the article.
+
+## ✍️ Reflection
```
Comment (emsesc) [inline Slack-Apps/week1/homework/responses/1.1-bolt-intro.md:9]: I like asking the students about what they found interesting/difficult to understand, but I think asking them a specific question about the content would "force" them to think more deeply about their understanding. At least for me as a student, I find when teachers ask me to write down interesting topics etc. I pick 3 random things and call it a day.

tl;dr Ask students to present a use case of Slack they would be interested in learning more about and how a piece of code from the samples could apply to this idea. Afterwards, ask them to describe any issues they had with comprehending the content--ie. do you have any questions?

File: Slack-Apps/week1/homework/responses/1.2-bots-and-apps.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+## 🧠 Brainstorm
+
+The article on bots listed a couple of their use cases, including:
+
+* Sending DMs!
+* Posting messages!
+* Uploading files!
+
+Come up with 2-3 ideas for a Slack app and 2 ideas for a bot and leave them as comments. These ideas don't have to consider any logistics, and can be as simple or complex as you would like!
```
Comment (emsesc) [inline Slack-Apps/week1/homework/responses/1.2-bots-and-apps.md:17]: I like this reflection/comment to move on topic 👍
Maybe also ask them to describe what problem one of their app/bots would solve?

File: Slack-Apps/week1/homework/responses/1.1-bolt-intro.md (hunk 1 above)
Comment (emsesc) [inline Slack-Apps/week1/homework/responses/1.1-bolt-intro.md:9]: I like asking the students about what they found interesting/difficult to understand, but I think asking them a specific question about the content would "force" them to think more deeply about their understanding. At least for me as a student, I find when teachers ask me to write down interesting topics etc. I pick 3 random things and call it a day.

tl;dr Ask students to present a use case of Slack they would be interested in learning more about and how a piece of code from the samples could apply to this idea. Afterwards, ask them to describe any issues they had with comprehending the content--ie. do you have any questions?

File: Slack-Apps/week1/homework/responses/1.3-first-app-and-events.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Your First Slack App
+
+Now that you have done some research on Slack development, its time to get coding! 
+
+## 🛠 Building your App
+
+You will be building your [first app](https://slack.dev/bolt-js/tutorial/getting-started#:~:text=%20Getting%20started%20with%20Bolt%20for%20JavaScript%20,handled%2C%20it%E2%80%99s%20time%20to%20set%20up...%20More%20%E2%80%A9https://slack.dev/bolt-js/tutorial/getting-started), which is a bot that listens for and responds to specific text in chats. **Note:** in order to make a public facing url, you first will need to [install ngrok](https://api.slack.com/tutorials/tunneling-with-ngrok), which dedicates a server on your end to receiving requests from the slack app. 
+
+While you're following both tutorials, write down any questions you have (ie what is this dependency, what is a scope). Then, research one of those questions and submit a 2-4 sentence summary of your findings as a comment.
```
Comment (emsesc) [inline Slack-Apps/week1/homework/responses/1.3-first-app-and-events.md:9]: What should the students do if they don't have any questions about the content?

File: Slack-Apps/week1/homework/responses/1.3-first-app-and-events.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+While you're following both tutorials, write down any questions you have (ie what is this dependency, what is a scope). Then, research one of those questions and submit a 2-4 sentence summary of your findings as a comment. 
+
+After you have completed the project, try changing what the app listens for (ie instead of "hello" it checks for "bit camp"). See if there is anything you can't listen for as part of a chat. 
+
+## 🔔 Learning about Events
+
+A large part of the app you just built is a **slack event**. Understanding events will be important as we will be using them throughout the bootcamp. Skim through the following articles to learn more about what an event is in programming, how events are used with the Slack Events API, and types of events that you can use in a project:
+
```
Comment (emsesc) [inline Slack-Apps/week1/homework/responses/1.3-first-app-and-events.md:16]: maybe think about adding in videos/visuals? depending on the student, they might prefer watching something over reading articles

File: Slack-Apps/week1/homework/responses/1.3-first-app-and-events.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+Now that you have done some research on Slack development, its time to get coding! 
+
+## 🛠 Building your App
+
+You will be building your [first app](https://slack.dev/bolt-js/tutorial/getting-started#:~:text=%20Getting%20started%20with%20Bolt%20for%20JavaScript%20,handled%2C%20it%E2%80%99s%20time%20to%20set%20up...%20More%20%E2%80%A9https://slack.dev/bolt-js/tutorial/getting-started), which is a bot that listens for and responds to specific text in chats. **Note:** in order to make a public facing url, you first will need to [install ngrok](https://api.slack.com/tutorials/tunneling-with-ngrok), which dedicates a server on your end to receiving requests from the slack app. 
+
+While you're following both tutorials, write down any questions you have (ie what is this dependency, what is a scope). Then, research one of those questions and submit a 2-4 sentence summary of your findings as a comment. 
+
+After you have completed the project, try changing what the app listens for (ie instead of "hello" it checks for "bit camp"). See if there is anything you can't listen for as part of a chat.
```
Comment (emsesc) [inline Slack-Apps/week1/homework/responses/1.3-first-app-and-events.md:11]: This is being picky, but maybe make this more clear by differentiating between the two tasks. Say something like:
> After you've completed the project:
> * Try changing what the app listens for (ie instead of "hello" it checks for "bit camp")
> * Play around with the code and see if there is anything you can't listen for as part of a chat. 

How will mentors know if they completed the app/have any problems with it? Will they commit the code?

**Review by emsesc — COMMENTED** [review]
Overall, great content - just some picky stuff.
